server/utils/oauthGoogleStrategy.js

```diff
@@ -1,21 +1,34 @@
-const passport = require('passport')
 const GoogleStrategy = require('passport-google-oauth2').Strategy;
+const passport = require('passport');
+const User = require('../models/User');
 
-passport.use(new GoogleStrategy({
-    clientID: process.env.GOOGLE_CLIENT_ID,
-    clientSecret: process.env.GOOGLE_CLIENT_SECRET,
-    callbackURL: process.env.CALL_BACK_URL,
-    passReqToCallback: true
-  },
-  function(request, accessToken, refreshToken, profile, done) {
-       return done(null, profile)
-  }
-));
+passport.use(
+	new GoogleStrategy(
+		{
+			clientID: process.env.GOOGLE_CLIENT_ID,
+			clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+			callbackURL: '/auth/google/callback',
+			passReqToCallback: true,
+			proxy: true
+		},
+		async function (request, accessToken, refreshToken, profile, done) {
+			const existingUser = await User.findOne({ email: profile.email });
+
+			// If the user already has an account, return their data
```

Comment:
Just a tip on comments, the code itself explains well so these kind of comments are not necessary.